File: solutions/_1343_number_of_sub_arrays_of_size_k_and_average_greater_than_or_equal_to_threshold/solution.py
from typing import List


class Solution:
    def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:
        count = 0
        cur_sum = 0
        i = 0
        j = 0

        while j < len(arr):
            cur_sum += arr[j]
            if j - i + 1 == k:
                if cur_sum / k >= threshold:
                    count += 1
                cur_sum -= arr[i]
                i += 1
            j += 1

        return count


# https://leetcode.com/problems/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/

File: solutions/_1343_number_of_sub_arrays_of_size_k_and_average_greater_than_or_equal_to_threshold/solution.java

package _1343_number_of_sub_arrays_of_size_k_and_average_greater_than_or_equal_to_threshold;

class Solution {
    public int numOfSubarrays(int[] arr, int k, int threshold) {
        int count = 0;
        int sum = 0;
        int i = 0;
        int j = 0;
        while (j < arr.length) {
            sum += arr[j];
            if (j - i + 1 == k) {
                if (sum / k >= threshold) {
                    count++;
                }
                sum -= arr[i];
                i++;
            }
            j++;
        }
        return count;
    }
}

/*
 * https://leetcode.com/problems/number-of-sub-arrays-of-size-k-and-average-
 * greater-than-or-equal-to-threshold/
 */